test/CodeGeneration/CUDACtorDtor.C

```diff
@@ -0,0 +1,62 @@
+//------------------------------------------------------------------------------
+// CLING - the C++ LLVM-based InterpreterG :)
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+//------------------------------------------------------------------------------
+
+// The Test checks, if the symbols __cuda_module_ctor and __cuda_module_dtor are
+// unique for every module. Attention, for a working test case, a cuda 
+// fatbinary is necessary.
+// RUN: cat %s | %cling -x cuda -Xclang -verify 2>&1 | FileCheck %s
+// REQUIRES: cuda-runtime
+
+#include "cling/Interpreter/Interpreter.h"
+#include "cling/Interpreter/Transaction.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+#include <iostream>
+
+// It compares the cuda module ctor and dtor of two random modules.
+std::string ctor1, ctor2, dtor1, dtor2;
+
+auto M = gCling->getLatestTransaction()->getModule();
+
+for(auto I = M->begin(), E = M->end(); I != E; ++I){
+  if((*I).getName().startswith_lower("__cuda_module_ctor_")){
```

Comment:
Why does this test for a trailing `'_'` while the code in the pass does not?

Reply:
The original name of the function is `__cuda_module_ctor` or `__cuda_module_dtor`, which the pass should detect. If it detects, it append an '_' and the module name.

In the test case, I check, if the '_' exist, because it is a sign, that the name was modified. I also could get the name of module, append to the search string and use equals instead startswith. 

At the moment, it is a simplified test, to check if the function name was modified. I also can write a more complicated and safer test method.

Reply:
Thanks, makes perfect sense. Please simply add a comment in the test, explaining why you test for the extra trailing `'_'`.